runtime_lib/CMakeLists.txt

```diff
@@ -5,81 +5,86 @@
-
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/me_basic.o DESTINATION ${CMAKE_INSTALL_PREFIX}/runtime_lib/${arch})
-
-
-  set(INSTALLS
-      chess_intrinsic_wrapper.cpp)
-
-  foreach(file ${INSTALLS})
-      add_custom_target(aie-copy-${arch}-runtime-libs-${file} ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${file})
-      add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}//${file}
-                      COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/${file}
-                      ${CMAKE_CURRENT_BINARY_DIR}/${file}
-                      DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${file})
-      add_dependencies(runtime-libs aie-copy-${arch}-runtime-libs-${file} )
-  endforeach()
-
-  install(FILES ${INSTALLS} DESTINATION ${CMAKE_INSTALL_PREFIX}/runtime_lib/${arch})
-
-  add_subdirectory(aiesim)
+# cmake needs this line
+cmake_minimum_required(VERSION 3.20.1)
 
-endfunction()
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-set(INSTALLS
-    test_library.cpp
-    test_library.h)
+include(ExternalProject)
 
-# Stuff into the build area:
-add_custom_target(runtime-libs ALL)
+project("runtime-libs")
+#add_custom_target(runtime-libs ALL)
 
-foreach(file ${INSTALLS})
-    add_custom_target(aie-copy-runtime-libs-${file} ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${file})
-    add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${file}
-                    COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/${file}
-                    ${CMAKE_CURRENT_BINARY_DIR}/${file}
-                    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${file})
-    add_dependencies(runtime-libs aie-copy-runtime-libs-${file} )
-endforeach()
+set(AIE_RUNTIME_TEST_TARGET_VAL ${AIR_RUNTIME_TEST_TARGET})
+set(x86_64_TOOLCHAIN_FILE "${CMAKE_CURRENT_SOURCE_DIR}/../cmake/toolchainFiles/toolchain_x86_64.cmake")
+if(SysrootAarch64)
+set(aarch64_TOOLCHAIN_FILE "${CMAKE_CURRENT_SOURCE_DIR}/../cmake/toolchainFiles/toolchain_clang_crosscomp_pynq.cmake")
+else()
+set(aarch64_TOOLCHAIN_FILE "${CMAKE_CURRENT_SOURCE_DIR}/../cmake/modulesXilinx/toolchain_vitis_crosscomp_arm.cmake")
+endif()
 
-# Install too
-install(FILES ${INSTALLS} DESTINATION ${CMAKE_INSTALL_PREFIX}/runtime_lib)
+if (DEFINED VITIS_ROOT)
+foreach(target ${AIE_RUNTIME_TARGETS})
+  # By default, we test the first architecture in AIE_RUNTIME_TARGETS.
+  # Alternatively, this can be defined to force testing with a particular architecture.
+  if (NOT AIE_RUNTIME_TEST_TARGET_VAL)
+    set(AIE_RUNTIME_TEST_TARGET_VAL ${target})
+    message("Testing with AIE runtime target: ${AIE_RUNTIME_TEST_TARGET_VAL}")
+  endif()
 
-add_subdirectory(AIE)
-add_subdirectory(AIE2)
+  if (NOT EXISTS ${${target}_TOOLCHAIN_FILE})
+    message(FATAL_ERROR "Toolchain file ${${target}_TOOLCHAIN_FILE} not found! Cannot build target ${target}.")
+  endif()
+  message("Building AIE runtime for ${target} with TOOLCHAIN=${${target}_TOOLCHAIN_FILE}")
+  
+  # xaiengine
+  ExternalProject_Add(xaiengine_${target}
```

Comment:
@denolf Would really like to avoid having 2 ExternalProjects here.

Reply:
@stephenneuendorffer since these are two different targets that need to get cross compiled, they need their own external project. As soon as 2023.1 is released, I plan to no longer get the code from the Vitis environment but from https://github.com/Xilinx/aie-rt/ and that will also motivates having a separate external project for xaiengine.

Reply:
We don't really need an external project per sub-directory: it's quite heavyweight.  I think in this case, we're not using cmake stuff from the Vitis environment (we are providing all the cmake stuff).  Will that change with aie-rt?